tests/integrationtests/tst_perfparser.cpp

```diff
@@ -0,0 +1,269 @@
+        QVERIFY(searchForChildSymbol(m_bottomUpData.children.at(bottomUpTopIndex), "main"));
+
+        auto topDownTopResult = std::max_element(m_topDownData.children.begin(), m_topDownData.children.end());
+        int topDownTopIndex = std::distance(m_topDownData.children.begin(), topDownTopResult);
+        QVERIFY(searchForChildSymbol(m_topDownData.children.at(topDownTopIndex), "main"));
+    }
+
+    void testCppRecursionEventCycles()
+    {
+        QStringList perfOptions;
+        perfOptions << "--event" << "cycles" << "-F" << "999";
+        QStringList exeOptions;
+        exeOptions << "45";
+
+        const QString exePath = QDir::currentPath() + "/../test-clients/cpp-recursion/cpp-recursion";
+        m_outputPath = QDir::currentPath() + "/event-cycles-cpp-re-perf.data";
+
+        perfRecord(perfOptions, exePath, exeOptions);
+
+        testPerfData("fibonacci", "cpp-recursion", "fibonacci", "cpp-recursion");
+    }
+
+private:
+    SummaryData m_summaryData;
+    Data::BottomUp m_bottomUpData;
+    Data::TopDown m_topDownData;
+    Data::CallerCalleeEntryMap m_callerCalleeData;
+    QString m_perfCommand;
+    QString m_outputPath;
+
+    void perfRecord(const QStringList &perfOptions, const QString& exePath, const QStringList &exeOptions)
+    {
+        PerfRecord perf(this);
+        QSignalSpy recordingFinishedSpy(&perf, &PerfRecord::recordingFinished);
+        QSignalSpy recordingFailedSpy(&perf, &PerfRecord::recordingFailed);
+
+        perf.record(perfOptions, m_outputPath, exePath, exeOptions);
+
+        while (recordingFinishedSpy.count() == 0 && recordingFailedSpy.count() == 0 ) {
+            QTest::qWait(200);
+        }
+
+        QCOMPARE(recordingFailedSpy.count(), 0);
+        QCOMPARE(recordingFinishedSpy.count(), 1);
+        QCOMPARE(QFileInfo::exists(m_outputPath), true);
+
+        m_perfCommand = perf.getPerfCommand();
+    }
+
+    void testPerfData(QString topBottomUpSymbolName, QString topBottomUpBinaryName, QString topTopDownSymbolName, QString topTopDownBinaryName)
+    {
+        PerfParser parser(this);
+
+        QSignalSpy parsingFinishedSpy(&parser, &PerfParser::parsingFinished);
+        QSignalSpy parsingFailedSpy(&parser, &PerfParser::parsingFailed);
+        QSignalSpy summaryDataSpy(&parser, &PerfParser::summaryDataAvailable);
+        QSignalSpy bottomUpDataSpy(&parser, &PerfParser::bottomUpDataAvailable);
+        QSignalSpy topDownDataSpy(&parser, &PerfParser::topDownDataAvailable);
+        QSignalSpy callerCalleeDataSpy(&parser, &PerfParser::callerCalleeDataAvailable);
+
+        parser.startParseFile(m_outputPath, "", "", "", "", "", "");
+
+        while (parsingFinishedSpy.count() == 0 && parsingFailedSpy.count() == 0 ) {
+            QTest::qWait(200);
+        }
+
+        // Clean up
+        QFile outputFile(m_outputPath);
+        outputFile.remove();
+
+        QCOMPARE(parsingFailedSpy.count(), 0);
+        QCOMPARE(parsingFinishedSpy.count(), 1);
+
```

Comment:
here and below: please add some comments to the code, explaining what the test actually checks for from a birds-eye POV